stan/math/prim/arr/functor/map_rect_mpi.hpp

```diff
@@ -0,0 +1,147 @@
+#pragma once
+
+#include <vector>
+#include <set>
+
+#include <stan/math/prim/mat/fun/to_array_1d.hpp>
+#include <stan/math/prim/arr/functor/mpi_command.hpp>
+#include <stan/math/prim/arr/functor/mpi_cluster.hpp>
+
+namespace stan {
+  namespace math {
+
+    namespace internal {
+      // forward declare
+      void
+      distribute_map_rect_data(const std::vector<std::vector<double> >& x_r,
+                               const std::vector<std::vector<int> >& x_i,
+                               const std::size_t uid);
+
+
+      // data distribution needs to go to prim
```

Comment:
what does this mean?

Reply:
I think the data distribution should be inside prim. That mechanism should be independent of rev/fwd/whatever. It's prim, not more.

Reply:
Ah, yes. I read it as a yet-to-be-accomplished directive, oops.